src/remote/activitypub/audience.ts

```diff
@@ -0,0 +1,93 @@
+import { ApObject, getApIds } from './type';
+import Resolver from './resolver';
+import { resolvePerson } from './models/person';
+import { unique, concat } from '../../prelude/array';
+import * as promiseLimit from 'promise-limit';
+import { User, IRemoteUser } from '../../models/entities/user';
+
+type Visibility = 'public' | 'home' | 'followers' | 'specified';
+
+type AudienceInfo = {
+	visibility: Visibility,
+	mentionedUsers: User[],
+	visibleUsers: User[],
+};
+
+export async function parseAudience(actor: IRemoteUser, to: ApObject | undefined, cc: ApObject | undefined, resolver?: Resolver): Promise<AudienceInfo> {
```

Comment:
```suggestion
export async function parseAudience(actor: IRemoteUser, to?: ApObject, cc?: ApObject, resolver?: Resolver): Promise<AudienceInfo> {
```

もし省略不能を意識させるのを意図した設計なら `null` の方がいいかもと思ったけど `getApIds` も触る必要が出てくるので、もし前述の通りであればとりあえず無視して Resolve してください。

Reply:
`なんたら | undefined` は`type`だとlintに指摘されるのに`function`だと指摘してくれない…